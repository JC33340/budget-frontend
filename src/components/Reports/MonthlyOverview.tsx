import ItemWrapper from '../ItemWrapper';
import ItemHeading from '../ItemHeading';
import type { reportsDataType } from '../../pages/Reports';
import MonthItem from './MonthItem';

type MonthlyOverviewType = {
    data?: reportsDataType[];
};

const MonthlyOverview = ({ data = [] }: MonthlyOverviewType) => {
    return (
        <ItemWrapper>
            <div className="flex flex-col gap-y-4">
                <ItemHeading text="Overview" />
                <div className="flex gap-x-4 overflow-x-scroll flex-nowrap ">
                    {data.length > 1 ? (
                        data.map((item, i) => <MonthItem key={i} item={item} />)
                    ) : (
                        <div className="w-full text-center font-semibold">
                            No data
                        </div>
                    )}
                </div>
            </div>
        </ItemWrapper>
    );
};

export default MonthlyOverview;
