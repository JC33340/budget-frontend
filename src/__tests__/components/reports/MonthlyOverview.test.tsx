import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import type { reportsDataType } from '../../../pages/Reports';
import MonthlyOverview from '../../../components/Reports/MonthlyOverview';

const testObj: reportsDataType = {
    year: 2025,
    month: 11,
    balance: 999,
    income: 39,
    expense: 56,
    categories: { testCat: 790 },
};

test('snapshot test', () => {
    render(<MonthlyOverview data={[testObj, testObj]} />);
    const item = screen.getByTestId('monthlyOverview-testid');
    expect(item).toMatchSnapshot();
});

test('content test', () => {
    render(<MonthlyOverview data={[testObj, testObj]} />);
    //testing that 2 of the children items have been rendered
    const children = screen.getAllByTestId('monthItem-testid');
    expect(children).toHaveLength(2);
});
