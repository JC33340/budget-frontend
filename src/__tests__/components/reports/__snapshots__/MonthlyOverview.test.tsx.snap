// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  class="flex flex-col gap-y-4"
  data-testid="monthlyOverview-testid"
>
  <p
    class="text-3xl text-blue font-medium text-center"
    data-testid="itemHeading-testid"
  >
    Overview
  </p>
  <div
    class="flex gap-x-4 overflow-x-scroll flex-nowrap "
  >
    <div
      class="p-2 border-2 border-dark-gray rounded-md w-full whitespace-nowrap"
      data-testid="monthItem-testid"
    >
      <p
        class="text-center font-semibold underline text-lg"
      >
        December
         
        <span>
          (
          2025
          )
        </span>
      </p>
      <p>
        <span
          class="font-semibold"
        >
          Total Balance:
        </span>
         
        <span>
          999
        </span>
      </p>
      <p
        class="text-green"
      >
        <span
          class="font-semibold"
        >
          Income:
        </span>
         
        <span>
          39
        </span>
      </p>
      <p
        class="text-red"
      >
        <span
          class="font-semibold"
        >
          Expense:
        </span>
         
        <span>
          56
        </span>
      </p>
      <div>
        <p
          class="font-semibold"
        >
          Category Breakdown:
        </p>
        <div
          class="max-h-[5rem] overflow-y-scroll border-1 border-light-gray px-2 py-1 rounded-md"
        >
          <div
            class="capitalize"
          >
            testCat
            : 
            790
          </div>
        </div>
      </div>
    </div>
    <div
      class="p-2 border-2 border-dark-gray rounded-md w-full whitespace-nowrap"
      data-testid="monthItem-testid"
    >
      <p
        class="text-center font-semibold underline text-lg"
      >
        December
         
        <span>
          (
          2025
          )
        </span>
      </p>
      <p>
        <span
          class="font-semibold"
        >
          Total Balance:
        </span>
         
        <span>
          999
        </span>
      </p>
      <p
        class="text-green"
      >
        <span
          class="font-semibold"
        >
          Income:
        </span>
         
        <span>
          39
        </span>
      </p>
      <p
        class="text-red"
      >
        <span
          class="font-semibold"
        >
          Expense:
        </span>
         
        <span>
          56
        </span>
      </p>
      <div>
        <p
          class="font-semibold"
        >
          Category Breakdown:
        </p>
        <div
          class="max-h-[5rem] overflow-y-scroll border-1 border-light-gray px-2 py-1 rounded-md"
        >
          <div
            class="capitalize"
          >
            testCat
            : 
            790
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
